test/source.test.ts

```diff
@@ -84,6 +84,86 @@ test("source", async () => {
   expect(loadHit).toEqual(["x", "x"]);
 });
 
+test("source should load once when multiple loads for the same source are triggered", async () => {
+  const Item = types
+    .model("Item", {
+      id: types.identifierNumber,
+      text: types.string,
+      feature: types.string
+    })
+    .views(self => ({
+      get displayText() {
+        return "Display " + self.text;
+      }
+    }));
+
+  const Container = types.model("Container", {
+    entryMap: types.map(Item)
+  });
+
+  const container = Container.create({ entryMap: {} });
+
+  const data = [
+    { id: 1, text: "A", feature: "x" },
+    { id: 2, text: "B", feature: "x" },
+    { id: 3, text: "C", feature: "y" }
+  ];
+
+  let loadHit: number = 0;
+
+  const load = async ({ feature }: { feature: string }) => {
+    loadHit++;
+    return data;
+  };
+
+  const source = new Source({
+    entryMap: container.entryMap,
+    load,
+    cacheDuration: 2
+  });
+
+  await Promise.allSettled([
```

Comment:
I was wondering whether allSettled is really needed here, or whether you couldn't get away with all()?

Reply:
I wanted to simulate a simultaneous trigger of load, like loading a lot of widgets at the same time would do.
`all()` would just fire them all in order and awaiting each one of them. `allSettled` just fires all of them at the same time and returns when all of them are finished.